x-pack/legacy/plugins/siem/public/containers/case/api.ts

```diff
@@ -47,62 +60,68 @@ export const getCases = async ({
   const tags = [...(filterOptions.tags?.map(t => `case-workflow.attributes.tags: ${t}`) ?? [])];
   const query = {
     ...queryParams,
-    filter: tags.join(' AND '),
-    search: filterOptions.search,
+    ...(tags.length > 0 ? { filter: tags.join(' AND ') } : {}),
+    ...(filterOptions.search.length > 0 ? { search: filterOptions.search } : {}),
   };
-  const response = await KibanaServices.get().http.fetch(`${CASES_URL}`, {
+  const response = await KibanaServices.get().http.fetch<CasesResponse>(`${CASES_URL}/_find`, {
     method: 'GET',
     query,
     asResponse: true,
   });
   await throwIfNotOk(response.response);
-  return convertAllCasesToCamel(response.body!);
+  return convertAllCasesToCamel(decodeCasesResponse(response.body));
 };
 
-export const createCase = async (newCase: NewCase): Promise<Case> => {
-  const response = await KibanaServices.get().http.fetch(`${CASES_URL}`, {
+export const createCase = async (newCase: CaseRequest): Promise<Case> => {
+  const response = await KibanaServices.get().http.fetch<CaseResponse>(`${CASES_URL}`, {
     method: 'POST',
     asResponse: true,
     body: JSON.stringify(newCase),
   });
   await throwIfNotOk(response.response);
-  return convertToCamelCase<CaseSnake, Case>(response.body!);
+  return convertToCamelCase<CaseResponse, Case>(decodeCaseResponse(response.body));
 };
 
 export const updateCaseProperty = async (
   caseId: string,
-  updatedCase: Partial<Case>,
+  updatedCase: Partial<CaseRequest>,
   version: string
-): Promise<Partial<Case>> => {
+): Promise<Case> => {
   const response = await KibanaServices.get().http.fetch(`${CASES_URL}/${caseId}`, {
     method: 'PATCH',
     asResponse: true,
-    body: JSON.stringify({ case: updatedCase, version }),
+    body: JSON.stringify({ ...updatedCase, id: caseId, version }),
   });
   await throwIfNotOk(response.response);
-  return convertToCamelCase<Partial<CaseSnake>, Partial<Case>>(response.body!);
+  return convertToCamelCase<CaseResponse, Case>(decodeCaseResponse(response.body));
 };
 
-export const createComment = async (newComment: NewComment, caseId: string): Promise<Comment> => {
-  const response = await KibanaServices.get().http.fetch(`${CASES_URL}/${caseId}/comment`, {
-    method: 'POST',
-    asResponse: true,
-    body: JSON.stringify(newComment),
-  });
+export const createComment = async (
+  newComment: CommentRequest,
+  caseId: string
+): Promise<Comment> => {
+  const response = await KibanaServices.get().http.fetch<CommentResponse>(
+    `${CASES_URL}/${caseId}/comments`,
+    {
+      method: 'POST',
+      asResponse: true,
+      body: JSON.stringify(newComment),
+    }
+  );
   await throwIfNotOk(response.response);
-  return convertToCamelCase<CommentSnake, Comment>(response.body!);
+  return convertToCamelCase<CommentResponse, Comment>(decodeCommentResponse(response.body));
 };
 
 export const updateComment = async (
```

Comment:
Can we rename this to `patchComment` to match the API?

Reply:
the API is returning the full comment